server/src/main/java/org/opensearch/search/rescore/QueryRescorerBuilder.java

```diff
@@ -190,8 +192,10 @@ public static QueryRescorerBuilder fromXContent(XContentParser parser) throws IO
     @Override
     public QueryRescoreContext innerBuildContext(int windowSize, QueryShardContext context) throws IOException {
         QueryRescoreContext queryRescoreContext = new QueryRescoreContext(windowSize);
-        // query is rewritten at this point already
-        queryRescoreContext.setQuery(queryBuilder.toQuery(context));
+
+        ParsedQuery parsedQuery = context.toQuery(queryBuilder);
+        queryRescoreContext.setParsedQuery(parsedQuery);
```

Comment:
a nit, lets one line this like before `queryRescoreContext.setParsedQuery(context.toQuery(queryBuilder));`